Documentation/faq.md

```diff
@@ -8,10 +8,50 @@
 
 `advertise-urls` specifies the addresses etcd clients or other etcd members should use to contact the etcd server. The advertise addresses must be reachable from the remote machines. Do not advertise addresses like `localhost` or `0.0.0.0` for a production setup since these addresses are unreachable from remote machines.
 
+### Deployment
+
+#### Why odd number of cluster size?
+
+A quorum is the majority of nodes in a cluster. When n is the cluster size, quorum is (n/2)+1. 2 is the quorum both for 3 and 4. The primary goal of etcd is provide reliability and availability. etcd requires the majority (`n/2 + 1` when `n` is the cluster size) to be available to operate. Thus, adding another node to 3-node cluster does not add any value in terms of fault tolerance. And adding 1 node to single-node cluster increases the quorum size and causes leader election; if a wrong node were added by mistake, the cluster would lose its leader and not be able to make any progress. Always start with an odd number of nodes.
+
+#### What is maximum cluster size?
+
+Theoretically, there is no hard limit. However, we recommend no more than 7 nodes. [Google Chubby lock service][chubby], which is a similar project to etcd and widely deployed within Google for many years, suggests running 5 nodes. A 5-member etcd cluster can provide enough fault tolerance in most cases. While larger cluster provides better fault tolerance the write performance reduces since data needs to be replicated to more machines.
```

Comment:
`
Theoretically, there is no hard limit. However, an etcd cluster probably should have no more than seven nodes. [Google Chubby lock service][chubby], similar to etcd and widely deployed within Google for many years, suggests running five nodes. A 5-member etcd cluster can tolerates two member failures, which is enough in most cases. Although larger clusters provide better fault tolerance, the write performance suffers because data must be replicated across more machines.
`